services/pull/check.go

```diff
@@ -32,7 +32,15 @@ func AddToTaskQueue(pr *models.PullRequest) {
 	go func() {
 		err := prQueue.PushFunc(strconv.FormatInt(pr.ID, 10), func() error {
 			pr.Status = models.PullRequestStatusChecking
-			err := pr.UpdateColsIfNotMerged("status")
+			divergence, err := GetDiverging(pr)
```

Comment:
Since task queues are parallelized and several commits could trigger this code, I believe there's a chance that the results might overwrite newer data.

Reply:
right but then we have a generall problem with other checks - or do I miss something?

Reply:
Yes, IMHO that's exactly the situation. Maybe you can find a good way to solve it here? 👼

Reply:
I think that can be left for other PR to deal with that problem globally